CMakeLists.txt

```diff
@@ -30,3 +30,33 @@ include(global_compile_options)
 
 # sub-projects
 add_subdirectory(sdk/core/azure-core)
+
+# tests
+enable_testing ()
+add_subdirectory(sdk/core/azure-core/test)
+
+#GTest
+# Download and unpack googletest at configure time
+configure_file(CMakeLists.txt.in googletest-download/CMakeLists.txt)
```

Comment:
This is great. We should add steps to the contributing guide/docs about how to run the tests as well.